app/kuma-dp/pkg/dataplane/readiness/component.go

```diff
@@ -0,0 +1,105 @@
+package readiness
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"sync/atomic"
+	"time"
+
+	"github.com/bakito/go-log-logr-adapter/adapter"
+
+	"github.com/kumahq/kuma/pkg/core"
+	"github.com/kumahq/kuma/pkg/core/runtime/component"
+)
+
+const (
+	pathPrefixReady  = "/ready"
+	stateReady       = "READY"
+	stateTerminating = "TERMINATING"
+)
+
+// Reporter reports the health status of this Kuma Dataplane Proxy
+type Reporter struct {
+	localListenPort uint32
+	isTerminating   atomic.Bool
+}
+
+var (
+	logger      = core.Log.WithName("readiness")
+	localIPAddr = "127.0.0.1"
```

Comment:
Here we need to support ipv4 and ipv6 (I believe)

Reply:
We don't support IPv6 only as of now, so either IPv4 or DualStack. So both of them include ipv4 address.

Reply:
Well, I can see the `AdminAddress`, we'd built such support. If I reuse the `AdminAddress` to be the `localIPAddr` here for the readiness reporter to listen, do you think that make sense?

Reply:
This is done now, please take a look at the lastest commit. Thank you.